File: parent-group4/pingyougou-interface/src/main/java/com/itheima/core/ItemCatCheckService.java
package com.itheima.core;

import com.itheima.core.pojo.item.ItemCat;
import com.itheima.core.pojo.item.ItemCatCheck;
import entity.PageResult;

import java.util.List;

public interface ItemCatCheckService {
    List<ItemCatCheck> findByParentId(Long parentId, String name);

    PageResult search(Integer page, Integer rows, Long parentId, String name);

    void add(ItemCatCheck itemCat, String name);

    void update(ItemCatCheck itemCat);

    void delete(Long[] ids);

    ItemCatCheck findOne(Long id);

    List<ItemCatCheck> findAll();

    PageResult search(Integer page, Integer rows);

    void updateStatus(Long[] ids, String status);
}

File: parent-group4/pingyougou-web-manager/src/main/webapp/js/service/itemCatCheckService.js
// 定义服务层:
app.service('itemCatCheckService',function($http){
    //读取列表数据绑定到表单中
    this.findAll=function(){
        return $http.get('../itemCatCheck/findAll.do');
    }
    this.findPage=function(page,rows){
        return $http.get('../itemCatCheck/findPage.do?page='+page+'&rows='+rows);
    }
    //删除
    this.dele=function(ids){
        return $http.get('../itemCatCheck/delete.do?ids='+ids);
    }
    //查询
    this.search=function(page,rows){
        return $http.get('../itemCatCheck/search.do?page='+page+"&rows="+rows);
    }
    //审核
    this.updateStatus = function(ids,status){
        return $http.get('../itemCatCheck/updateStatus.do?ids='+ids+"&status="+status);
    }

});